src/tools/rustfmt/src/types.rs

```diff
@@ -537,28 +537,19 @@ impl Rewrite for ast::Lifetime {
 impl Rewrite for ast::GenericBound {
     fn rewrite(&self, context: &RewriteContext<'_>, shape: Shape) -> Option<String> {
         match *self {
-            ast::GenericBound::Trait(ref poly_trait_ref, trait_bound_modifier) => {
+            ast::GenericBound::Trait(ref poly_trait_ref, modifiers) => {
```

Comment:
The rustfmt changes might be a bit controversial…

Reply:
@fmease I know this has already been merged, but please avoid making unnecessary changes to subtrees like rustfmt. It would have been more appropriate to open a PR for this directly in rustfmt. I would have asked for this to be reverted had I seen it earlier but given the holiday season this one slipped through.

cc: @compiler-errors @calebcartwright

Reply:
Sorry, but I don't understand what you mean by "unnecessary changes" in this particular case. I'm not sure how this PR could've been done without modifying rustfmt -- that's just part of the reality of rustfmt using data structures from `rustc_ast`.

Reply:
I'm certain @fmease or I will be happy to assist with a subtree sync if this ends up causing conflicts the next time you perform one.

Reply:
Maybe I don't fully understand the AST changes, but could the diff have been minimized? It seems like a lot larger of a diff than most other AST modifications.

Reply:
I don't believe this could be minimized in a way that doesn't affect code quality. This PR completely replaces the `TraitBoundModifier` enum.

What's the concern here? Synchronization conflicts? Do you expect this code to have been modified on the rustfmt side?

Reply:
I Appreciate you offering to help out with the sync if there are issue. I'm trying to get a subtree push done and this change jumped out as odd to me. It's not causing any issues, but it just looked like a much larger change than I would have expected. If this was the only way to rewrite the code then that's fair.

Reply:
I'm glad it's not causing you too much trouble. While this diff might seem large, the point of this refactoring PR was to eliminate boilerplate like the one in rustfmt where one had to explicitly match on each kind to obtain a textual representation.

Thanks for reaching out! I won't make a habit of making such kind of to rustfmt from within the rust repo ;)

In case you do need any help with the sync or something similar, I'm definitely available and happy to help :)

Reply:
Thanks for the clarification. I think I was quick to judge this as an unnecessary change. After taking another look the changes seem fine. Very much appreciate you offering to help in case there is an issue 🙏🏼